src/hints.rs

```diff
@@ -284,43 +283,72 @@ impl<'a> From<&'a NotificationHint> for MessageItem {
             NotificationHint::Invalid                  => ("invalid"      .to_owned(), MessageItem::Str("Invalid".to_owned()))
         };
 
-        MessageItem::DictEntry(
-            Box::new(hint.0.into()),
-            Box::new(MessageItem::Variant( Box::new(hint.1) ))
-            )
+        (MessageItem::Str(key), value)
```

Comment:
this is actually not quite correct. I don't know if those `realworld` test ran through on your desktop, but on my KDE I get a dbus Error: `Error { kind: Dbus(D-Bus error: No such method 'Notify' in interface 'org.freedesktop.Notifications' at object path '/org/freedesktop/Notifications' (signature 'susssasa{ss}i') (org.freedesktop.DBus.Error.UnknownMethod)) }` in the tests
`realworld::category`
`realworld::persistent`
`realworld::urgency`

admittedly those tests kinda ate the errors with the `assert!(....is_ok())` in those situations plain old `.unwrap()` works best. I made the output a bit more useful on this [branch](https://github.com/hoodie/notify-rust/tree/feature/breaking-bad-4.0).
pro tip: you can see the difference if you run `cargo test realworld::category` for this and master and watch what happens with `dbus-monitor "interface=org.freedesktop.Notifications"`

I think it might be fixed with `(MessageItem::Str(key), MessageItem::Variant(Box::new(value)))` instead.

Reply:
what desktop env did you test this on?